blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -134,29 +134,29 @@ builder.addGlobalStore(
+First, we designed [Kestra](https://github.com/kestra-io/kestra) to have only one **huge** stream for all the processing from the executor. Even though this seemed like an excellent choice initially, we would soon face some problems.
 
 
-Here is the last version of our main and only one Kafka Stream with many topics 🙉:
+Here is the last version of our main and only Kafka Stream with many topics 🙉:
 ![Kestra Topology](./2022-06-28-tips-kafka-streams-distributed/topology.jpg)
-Yes, this is some kind of very large Kafka Streams. It was working almost despite the complexity of this one. But the major drawback was :
-- **Monitoring**: all the metrics are under the same consumer groups
-- **Debugging**: during a crash, each topic is consumed independently, when a message failed, the whole process is crashed
-- and the most important is **lag**: since Kafka Stream optimizes the consumption of messages by themselves, a topic with large outputs could lead to lag on other topics unrelated topics. In that case, impossible to understand properly the lag on our consumer.
+Yes, this is a massive Kafka Stream. Despite its complexity, it worked well. However, there is a major drawback:
+- **Monitoring**: all the metrics are under the same consumer groups.
+- **Debugging**: during a crash, each topic is consumed independently; when a message fails, the whole process crushes.
+- **lag** (the most important): since Kafka Stream optimizes the consumption of messages by itself, a topic with large outputs could lead to lag on other unrelated topics. In that case, it is impossible to properly understand the lag on our consumer.
 
-Now, we have decided to split into multiple [streams](https://github.com/kestra-io/kestra/tree/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/executors) to be able to monitors and understand properly the lag on our Kafka Streams.
+Now, we have decided to split into multiple [streams](https://github.com/kestra-io/kestra/tree/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/executors) to be able to monitor and properly understand the lag on our Kafka Streams. 
 
-How to split your evil streams? We simply choose to consume only one time a topic (to avoid large network transit), so we grouped all streams by source topics.
+To split our bad streams, we simply choose to deal with only one topic at a time -- to avoid large network transit -- so we grouped all streams by source topics.
 
 
 ## Conclusion
 
-We have covered here some tips that took us a large amount of time to find the way to deal with the issue. [Kestra](https://github.com/kestra-io/kestra) is the only infinitely scalable data orchestration and scheduling platform, that used Kafka as backend, that allows millions of executions.
+From the insights above, we can now confidently conclude that Kafka was able to assist [Kestra](https://github.com/kestra-io/kestra) as a backend, making millions of executions possible.
```

Comment:
We could make the conclusion broader by explicitly indicating that even though there were some challenges, we were able to show that Kafka works well for our case scenario and others like it. This could also involve a summary of the main points we make in this article.